crates/air-lib/air-parser/src/parser/lexer/call_variable_parser.rs

```diff
@@ -387,12 +413,27 @@ impl MetTag {
     fn from_tag(tag: char) -> Self {
         match tag {
             '$' => Self::Stream,
-            '#' => Self::CanonStream,
+            '#' => Self::Canon,
             '%' => Self::StreamMap,
             _ => Self::None,
         }
     }
 
+    fn update_type(&self, tag: char) -> Self {
```

Comment:
IMHO should be `update_canon_type`.

Reply:
But most likely it will be context-dependent, won't it?

Reply:
Doesn't look like it. Match + if will do the trick.

Reply:
Then you will need to update all the call site, as you call it now under `if`.  Perhaps, it worth moving that check inside the function right now.

Reply:
This makes sense.

Reply:
Also, in this hypothetical future it is not so hard to rename the function back if it becomes more generic, is it?

Reply:
I prefer to stick with a generic name now to potentially save some "cycles" in future. I mean it looks like a matter of opinions.

Reply:
I will rename the function into maybe_change_type and move if into it.

Reply:
You should save reader's "cycles", not yours.

Reply:
The future programmer will be the reader/editor whose cycles I am saving. 
maybe_change_type will return a value thus a reader will see the asignment.

Reply:
IMHO, a method can update state and return something as well.